### PR TITLE
Get methods for vApp and VM

### PR DESCRIPTION
**Pull request by dataclouder**
This is the usual boilerplate to implement new Get methods.

* Add Get* methods for VM and VApp
* Deprecate methods FindVMByHREF, FindVMByName, FindVAppByID, and FindVAppByName
* Replace usage of deprecated Find* methods with new Get* methods

### CODE REVIEW

**Comment by Didainius** [comment]
And as usual - the changelog :)

**Comment by dataclouder** [comment]
> And as usual - the changelog :)

Done!

File: govcd/vapp.go [hunk 1]
```diff
@@ -858,3 +858,87 @@ func (vapp *VApp) SetProductSectionList(productSection *types.ProductSectionList
...
+	if err != nil {
+
+		return nil, err
+	}
+
+	return newVm, nil
+}
+
+// GetVMByName returns a VM reference if the VM name matches an existing one.
+// If no valid VM is found, it returns an empty VM reference and an error
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/vapp.go]: fixed

File: govcd/vapp.go [hunk 2]
```diff
@@ -858,3 +858,87 @@ func (vapp *VApp) SetProductSectionList(productSection *types.ProductSectionList
...
+			return vapp.client.GetVMByHref(child.HREF)
+		}
+
+	}
+	util.Logger.Printf("[TRACE] Couldn't find VM: %s", vmName)
+	return nil, ErrorEntityNotFound
+}
+
+// GetVMById returns a VM reference if the VM ID matches an existing one.
+// If no valid VM is found, it returns an empty VM reference and an error
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/vapp.go]: fixed

File: govcd/vapp.go [hunk 3]
```diff
@@ -858,3 +858,87 @@ func (vapp *VApp) SetProductSectionList(productSection *types.ProductSectionList
 func (vapp *VApp) GetProductSectionList() (*types.ProductSectionList, error) {
 	return getProductSectionList(vapp.client, vapp.VApp.HREF)
 }
+
+// GetVMByHref returns a VM reference by running a vCD API call
+// If no valid VM is found, it returns an empty VM reference and an error
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/vapp.go]: fixed

File: CHANGELOG.md [hunk 4]
```diff
@@ -31,6 +31,9 @@ vApp guest properties [#235](https://github.com/vmware/go-vcloud-director/pull/2
 * Added methods `CreateNsxvNatRule()`, `UpdateNsxvNatRule()`, `GetNsxvNatRuleById()`, `DeleteNsxvNatRuleById()`
 which use the proxied NSX-V API of advanced edge gateway for handling NAT rules [#241](https://github.com/vmware/go-vcloud-director/pull/241)
 * Added methods `GetVnicIndexByNetworkNameAndType()` and `GetNetworkNameAndTypeByVnicIndex()` [#241](https://github.com/vmware/go-vcloud-director/pull/241)
+* Added methods `Vdc.GetVappByHref`, `Vdc.GetVAppByName` and related `GetVAppById`, `GetVAppByNameOrId`
+* Added methods `Client.GetVMByHref` `Vapp.GetVAMByName` and related `GetVMById`, `GetVAMByNameOrId`
```
Comment (vbauzysvmware) [inline CHANGELOG.md]: `GetVAMByName`, ``GetVAMByNameOrId` - mistype `VAM`

**Pull request merged**